// @brojs/cli/lib/i18next.js
import { initReactI18next } from "react-i18next";
import Backend from "i18next-http-backend";
import LanguageDetector from 'i18next-browser-languagedetector';

const loadPath = () => `${__webpack_public_path__}locales/{{lng}}.json`;

const ajax = (url, opts, callback) =>
  fetch(url).then(resp => resp.json()).then((response) => {
    const enrichedLocales = {
      ...response.data,
    };
    callback(JSON.stringify(enrichedLocales), response);
  }).catch((e) => {
    callback('', e.response);
  });

export const i18nextInitConfig = (i18next) => i18next.use(Backend).init({
  lng: localStorage.getItem('i18nextLng') || "ru",
  fallbackLng: "ru",
  keySeparator: false,
  backend: {
    loadPath,
  },
});

export const i18nextReactInitConfig = (i18next) => i18next
  .use(Backend)
  .use(initReactI18next)
  .use(LanguageDetector)
  .init({
    fallbackLng: "ru",
    load: 'currentOnly',
    keySeparator: false,
    whitelist: ['ru', 'en'],
    backend: {
      loadPath,
    },
    interpolation: {
      escapeValue: false,
    }
  });
